src/patterns/gen-ai/aws-llama-index-data-loader/index.ts

```diff
@@ -0,0 +1,456 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+import { Duration, RemovalPolicy, Stack } from 'aws-cdk-lib';
+import { FlowLogDestination } from 'aws-cdk-lib/aws-ec2';
+import { DockerImageAsset } from 'aws-cdk-lib/aws-ecr-assets';
+import { Cluster, ContainerImage } from 'aws-cdk-lib/aws-ecs';
+import { QueueProcessingFargateService } from 'aws-cdk-lib/aws-ecs-patterns';
+import { Effect, PolicyStatement, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
+import { Key } from 'aws-cdk-lib/aws-kms';
+import { BlockPublicAccess, Bucket, BucketEncryption, EventType, ObjectLockMode } from 'aws-cdk-lib/aws-s3';
+import { SnsDestination } from 'aws-cdk-lib/aws-s3-notifications';
+import { Topic } from 'aws-cdk-lib/aws-sns';
+import { SqsSubscription } from 'aws-cdk-lib/aws-sns-subscriptions';
+import { Queue } from 'aws-cdk-lib/aws-sqs';
+import { StringParameter } from 'aws-cdk-lib/aws-ssm';
+import { NagSuppressions } from 'cdk-nag';
+import { Construct } from 'constructs';
+import { BaseClass, BaseClassProps, ConstructName } from '../../../common/base-class';
+
+export interface LlamaIndexDataLoaderProps {
+  /**
+   * The directory to build the Docker image
+   * @description The directory to build the Docker image.
+   */
+  readonly dockerImageAssetDirectory: string;
```

Comment:
as discussed here we can add a default dockerfile if none is provided

Reply:
done